src/sentry/api/event_search.py

```diff
@@ -357,9 +331,17 @@ def is_negated(self, node):
 
         return node.text == '!'
 
-    def visit_basic_filter(self, node, children):
-        negation, search_key, _, search_value = children
+    def visit_basic_filter(self, node, (negation, search_key, _, search_value)):
         operator = '!=' if self.is_negated(negation) else '='
+        return self._handle_basic_filter(search_key, operator, search_value)
+
+    def _handle_basic_filter(self, search_key, operator, search_value):
+        # If a date or numeric key gets down to the basic filter, then it means
+        # that the value wasn't in a valid format, so raise here.
+        if search_key.name in self.date_keys:
+            raise InvalidSearchQuery('Invalid format for date search')
+        if search_key.name in self.numeric_keys:
+            raise InvalidSearchQuery('Invalid format for numeric search')
```

Comment:
Should you tell the user which term is incorrect in case they provide multiple bad ones?

Reply:
Yeah, I've been thinking about doing so. It's tricky though, because we do a bunch of renaming of keys (`timesSeen -> times_seen`, `event.timestamp -> date`) and so on, and if we return the information as it currently is then it'll potentially be confusing to the user.

We probably need to store the original value for key/operator/value somewhere in the SearchFilter or the respective classes so that we can give this kind of feedback.

I'll make a separate ticket for doing so.